internal/store/build_result.go

```diff
@@ -13,94 +13,150 @@ import (
-	//
-	// When we deploy a Docker Compose service, we wait synchronously for the
-	// container to start. Note that this is a different concurrency model than
-	// we use for Kubernetes, where the pods appear some time later via an
-	// asynchronous event.
-	DockerComposeContainerID container.ID
+func (r ImageBuildResult) TargetID() model.TargetID { return r.id }
+func (r ImageBuildResult) BuildType() BuildType     { return BuildTypeImage }
+
+// For image targets.
+func NewImageBuildResult(id model.TargetID, image reference.NamedTagged) ImageBuildResult {
+	return ImageBuildResult{
+		id:    id,
+		Image: image,
+	}
+}
+
+type LiveUpdateBuildResult struct {
+	id model.TargetID
+
+	// The name+tag of the image that the pod is running.
+	Image reference.NamedTagged
 
 	// The ID of the container(s) that we live-updated in-place.
 	//
 	// The contents of the container have diverged from the image it's built on,
 	// so we need to keep track of that.
 	LiveUpdatedContainerIDs []container.ID
-
-	// The UIDs that we deployed to a Kubernetes cluster.
-	DeployedUIDs []types.UID
 }
 
+func (r LiveUpdateBuildResult) TargetID() model.TargetID { return r.id }
+func (r LiveUpdateBuildResult) BuildType() BuildType     { return BuildTypeLiveUpdate }
+
 // For in-place container updates.
-func NewLiveUpdateBuildResult(id model.TargetID, containerIDs []container.ID) BuildResult {
-	return BuildResult{
-		TargetID:                id,
+func NewLiveUpdateBuildResult(id model.TargetID, image reference.NamedTagged, containerIDs []container.ID) LiveUpdateBuildResult {
+	return LiveUpdateBuildResult{
+		id:                      id,
+		Image:                   image,
 		LiveUpdatedContainerIDs: containerIDs,
 	}
 }
 
-// For image targets.
-func NewImageBuildResult(id model.TargetID, image reference.NamedTagged) BuildResult {
-	return BuildResult{
-		TargetID: id,
-		Image:    image,
-	}
+type DockerComposeBuildResult struct {
+	id model.TargetID
+
+	// The ID of the container that Docker Compose created.
+	//
+	// When we deploy a Docker Compose service, we wait synchronously for the
+	// container to start. Note that this is a different concurrency model than
+	// we use for Kubernetes, where the pods appear some time later via an
+	// asynchronous event.
+	DockerComposeContainerID container.ID
 }
 
+func (r DockerComposeBuildResult) TargetID() model.TargetID { return r.id }
+func (r DockerComposeBuildResult) BuildType() BuildType     { return BuildTypeDockerCompose }
+
 // For docker compose deploy targets.
-func NewDockerComposeDeployResult(id model.TargetID, containerID container.ID) BuildResult {
-	return BuildResult{
-		TargetID:                 id,
+func NewDockerComposeDeployResult(id model.TargetID, containerID container.ID) DockerComposeBuildResult {
+	return DockerComposeBuildResult{
+		id:                       id,
 		DockerComposeContainerID: containerID,
 	}
 }
 
+type K8sBuildResult struct {
+	// The image target that this built.
```

Comment:
...wait, huh? wouldn't this be a k8s target ID?

Reply:
yep! removed